### ISSUE TITLE
Bagging with underlying probabilities

### ISSUE THREAD

**Issue by Prometheus77** [mlr-org/mlr#1920]
When I bag classification learners, I generally want to make probability predictions more accurate by having each model predict a probability and averaging those probabilities. However, bagging in MLR appears only to support predicting responses and then returning the proportion of models that predicted that response, which I expect is generally less likely to be accurate. Are there plans to allow bagging to support averaging of underlying probability predictions?

**Comment by larskotthoff** [comment]
The short answer is no. Why do you expect averaging probabilities to be more accurate?

**Comment by berndbischl** [comment]
what @Prometheus77 is suggesting is - of course - a reasonable request. IIRC there was a paper where they compared bagging the discrete labels with bagging the numeric probabilities. 
the results were inconclusive what is better. 

it should be an easy fix to do that. would you try to help?

**Comment by Prometheus77** [comment]
I now have a specific use case for this. I have an imbalanced class problem where false positives are expensive, and so I created a custom measure called "precision at 97.5%" which takes the top 2.5% of predicted probability and measures true positive rate in that batch. I want to tune an XGBoost model to this metric with a fixed out-of-time holdout sample, but the training set is too big for XGBoost to handle with my current memory configuration. A learning curve analysis on 30% of the training data (the most it can handle) shows a roughly linear improvement in performance with more data and no diminishing return up to 30%, so I want to make use of the entire set. I've settled on using bagging to tune XGBoost to optimize this measure, but since makeBaggingWrapper can only output 1 or 0 for per observation per bag, I don't have a reliable way to measure the top 2.5% predicted probability of each case.

Any chance of getting this prioritized? If not, I might try to hack it myself, but if as berndbischl states this is truly an easy fix, would prefer to see it done "right".
